### PR TITLE
database.apim_db.type is not complete. it should be support mysql,h2,mssql,postgre,oracle,db2

### PR DESCRIPTION
**Pull request by ibrohimislam**
`database.apim_db.type` entry on infer.json by @ruks and @bhathiya is not complete. it should be support mysql,h2,mssql,postgre,oracle,db2 as mentioned in https://apim.docs.wso2.com/en/latest/Reference/ConfigCatalog/.

there are two options:
- delete the entry `database.apim_db.type`. so it doesn't override `database.$1.type`;
- copy `database.$1.type` to this `database.apim_db.type`. so, it keep repo independency.

### CODE REVIEW

File: modules/distribution/product/src/main/resources/conf/infer.json [hunk 1]
```diff
@@ -76,9 +74,29 @@
...
+    },
+    "postgre": {
+      "database.apim_db.driver": "org.postgresql.Driver",
+      "database.apim_db.url": "jdbc:postgresql://$ref{database.apim_db.hostname}:$ref{database.apim_db.port}/$ref{database.apim_db.name}",
+      "database.apim_db.validationQuery": "SELECT 1; COMMIT"
+    },
     "h2": {
       "database.apim_db.driver": "org.h2.Driver",
-      "database.apim_db.url": "jdbc:h2:./repository/database/WSO2AM_DB;DB_CLOSE_ON_EXIT=FALSE;LOCK_TIMEOUT=60000",
+      "database.apim_db.url": "jdbc:h2:./repository/database/WSO2CARBON_DB;DB_CLOSE_ON_EXIT\u003dFALSE;LOCK_TIMEOUT\u003d60000",
```
Comment (bhathiya) [inline modules/distribution/product/src/main/resources/conf/infer.json]: What's the reason for this change?

Comment (ibrohimislam) [reply modules/distribution/product/src/main/resources/conf/infer.json]: it should be support mysql,h2,mssql,postgre,oracle,db2 as mentioned in https://apim.docs.wso2.com/en/latest/Reference/ConfigCatalog/.

Comment (ibrohimislam) [reply modules/distribution/product/src/main/resources/conf/infer.json]: oh, I just copy from `database.$1.type`.

File: modules/distribution/product/src/main/resources/conf/infer.json [hunk 2]
```diff
@@ -7,11 +7,9 @@
...
       "indexing.enable": false,
       "system.parameter.profile": "gateway-worker",
-      "enabled_global_handlers": [
-          "external_call_logger"
-        ],
-       "synapse_handlers.external_call_logger.name": "externalCallLogger",
-       "synapse_handlers.external_call_logger.class": "org.wso2.carbon.apimgt.gateway.handlers.LogsHandler"
+      "enabled_global_handlers": ["external_call_logger"],
+      "synapse_handlers.external_call_logger.name": "externalCallLogger",
+      "synapse_handlers.external_call_logger.class": "org.wso2.carbon.apimgt.gateway.handlers.LogsHandler"
```
Comment (ruks) [inline modules/distribution/product/src/main/resources/conf/infer.json]: seems no changes here and just formatting. can you revert it?

File: modules/distribution/product/src/main/resources/conf/infer.json (hunk 1 above)
Comment (ibrohimislam) [reply modules/distribution/product/src/main/resources/conf/infer.json]: I don't know where repo `database.$1.type` is come from. I just peek from running instance.

Comment (bhathiya) [reply modules/distribution/product/src/main/resources/conf/infer.json]: my question is why have you replaced `WSO2AM_DB` with `WSO2CARBON_DB `?

Comment (ibrohimislam) [reply modules/distribution/product/src/main/resources/conf/infer.json]: oh, sorry. it's a mistake.

Comment (ibrohimislam) [reply modules/distribution/product/src/main/resources/conf/infer.json]: it should be WSO2AM_DB

File: modules/distribution/product/src/main/resources/conf/infer.json (hunk 2 above)
Comment (ibrohimislam) [reply modules/distribution/product/src/main/resources/conf/infer.json]: oh, ok.

File: modules/distribution/product/src/main/resources/conf/infer.json [hunk 3]
```diff
@@ -76,9 +74,29 @@
...
       "database.apim_db.validationQuery": "SELECT 1"
     },
+    "oracle": {
+      "database.apim_db.driver": "oracle.jdbc.OracleDriver",
+      "database.apim_db.url": "jdbc:oracle:thin:@$ref{database.apim_db.hostname}:$ref{database.apim_db.port}/$ref{database.apim_db.sid}",
+      "database.apim_db.validationQuery": "SELECT 1 FROM DUAL"
+    },
+    "mssql": {
+      "database.apim_db.driver": "com.microsoft.sqlserver.jdbc.SQLServerDriver",
+      "database.apim_db.url": "jdbc:sqlserver://$ref{database.apim_db.hostname}:$ref{database.apim_db.port};databaseName\u003d$ref{database.apim_db.name};SendStringParametersAsUnicode\u003dfalse",
```
Comment (bhathiya) [inline modules/distribution/product/src/main/resources/conf/infer.json]: will it break if we replace `\u003d` with `=`?

Comment (ibrohimislam) [reply modules/distribution/product/src/main/resources/conf/infer.json]: I don't think so.

File: modules/distribution/product/src/main/resources/conf/infer.json [hunk 4]
```diff
@@ -76,9 +76,29 @@
...
+    },
+    "postgre": {
+      "database.apim_db.driver": "org.postgresql.Driver",
+      "database.apim_db.url": "jdbc:postgresql://$ref{database.apim_db.hostname}:$ref{database.apim_db.port}/$ref{database.apim_db.name}",
+      "database.apim_db.validationQuery": "SELECT 1; COMMIT"
+    },
     "h2": {
       "database.apim_db.driver": "org.h2.Driver",
-      "database.apim_db.url": "jdbc:h2:./repository/database/WSO2AM_DB;DB_CLOSE_ON_EXIT=FALSE;LOCK_TIMEOUT=60000",
+      "database.apim_db.url": "jdbc:h2:./repository/database/WSO2AM_DB;DB_CLOSE_ON_EXIT\u003dFALSE;LOCK_TIMEOUT\u003d60000",
```
Comment (bhathiya) [inline modules/distribution/product/src/main/resources/conf/infer.json]: do we need changes to this line at all? `=` character breaks anything?

File: modules/distribution/product/src/main/resources/conf/infer.json (hunk 3 above)
Comment (ibrohimislam) [reply modules/distribution/product/src/main/resources/conf/infer.json]: should it be changed to `=`, for readability?

Comment (bhathiya) [reply modules/distribution/product/src/main/resources/conf/infer.json]: yes, please.
